src/full_client.toit

```diff
@@ -278,20 +282,33 @@ abstract class DefaultReconnectionStrategyBase implements ReconnectionStrategy:
       [--reconnect_transport]
       [--send_connect]
       [--receive_connect_ack]:
-    for i := -1; i < attempt_delays_.size; i++:
+    attempt_counter := -1
+    while true:
       if is_closed: return null
-      is_last_attempt := (i == attempt_delays_.size - 1)
+
+      attempt_counter++
+
+      is_last_attempt := false
+      if attempt_delays_:
+        is_last_attempt = (attempt_counter == attempt_delays_.size)
 
       try:
         catch --unwind=(: is_closed or is_last_attempt):
-          logger_.debug "Attempting to connect"
 
-          if i == -1 and not reuse_connection:
-            reconnect_transport.call
-          else if i >= 0:
-            sleep_duration := attempt_delays_[i]
+          if attempt_counter == 0:
+            // In the first iteration we try to connect without delay.
+            if not reuse_connection:
+              logger_.debug "Attempting to reconnect"
```

Comment:
Is this already what you consider a reconnect (not just a connect)?

Reply:
Changed it to "(re)connect".